apps/wallet/src/ui/app/components/WalletListSelect.tsx

```diff
@@ -40,6 +41,14 @@ export function WalletListSelect({
         return accounts;
     }, [accounts, visibleValues]);
     const deriveNextAccount = useDeriveNextAccountMutation();
+    useEffect(() => {
```

Comment:
this doesn't feel like it should be an effect since it isn't a side effect of rendering, it's more so a result of someone clicking the "New account" button. I think we could remove this and just update `newAccounts` from where we call `mutate`